### ISSUE TITLE
numpy.ulong is missing

### ISSUE THREAD

**Issue by gmgunter** [numpy/numpy#21063]
It appears that NumPy doesn't define a scalar type `numpy.ulong`. This seems like a conspicuous omission -- given that the library exports many similar types such as `long` and `ulonglong`, I assumed that `ulong` would be a valid type.

Similarly, the following code snippet fails:

```python
>>> import numpy as np
>>> np.dtype("ulong")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: data type 'ulong' not understood
```

Is this a deliberate omission? I wasn't able to find any info about it. Is there any reason why we couldn't (or shouldn't) add `numpy.ulong` as an alias for `numpy.uint`?

**Comment by seberg** [comment]
I suppose it exists as `uint` (I know, I know, weird but `uintc` is the actual unsigned int).  But yea, we have `long` and that is useful for C analogy, so I agree with adding the alias.

**Comment by eric-wieser** [comment]
`np.long` doesn't exist any more, does it? I suppose we should still allow it as a dtype string, but exposing it globally is likely going to set downstream users up for breakage who are used to `np.long is int`.

**Comment by InessaPawson** [comment]
It appears that numpy.ulong is not defined deliberately. 
@seberg @eric-wieser Should we close this issue?

**Comment by seberg** [comment]
@InessaPawson yes, `np.ulong` is deliberate.  But it would be fine to support `np.dtype("ulong")` since `np.dtype("long")` exists.
(Not particularly high priority, but I would be in favor of merging such a PR/proposal – it is a simple change the only tricky part is finding where to change it.)

**Comment by gmgunter** [comment]
>it is a simple change the only tricky part is finding where to change it

I'd be happy to contribute a PR, but this is the issue I ran into. I can't find where in the codebase to make the necessary modification(s)!

Could you point me in the right direction? Otherwise, I'll just leave it up to the more experienced contributors.

**Comment by seberg** [comment]
@gmgunter it is a bit confusing/weird, but basically, that is the dictionary that does the mapping: https://github.com/numpy/numpy/blob/eeef9d4646103c3b1afd3085f1393f2b3f9575b2/numpy/core/_type_aliases.py#L26

and the code that fills it should be somewhere in the same file.
